packages/driver/cypress/e2e/commands/assertions.cy.js

```diff
@@ -648,6 +648,19 @@ describe('src/cy/commands/assertions', () => {
         cy.get('button:first', { timeout: 100 }).should('have.class', 'does-not-have-class')
       })
 
+      it('has a pending state while retrying for commands with onFail', (done) => {
+        cy.on('command:retry', (command) => {
+          const [readFileLog, shouldLog] = cy.state('current').get('logs')
```

Comment:
Seems like this has a chance to be flakey since this state will be updated as the command resolves and eventually change to 'pass'. Did you notice this as all when running locally?

Reply:
Shouldn't be - because we we call `done()` inside this retry, Cypress cancels all active commands and doesn't continue retrying commands. `cy.on('command:retry', () => { ...; done() }` is a pretty common pattern in the driver tests.